tracer/src/Datadog.AutoInstrumentation.NativeLoader/runtimeid_store.cpp

```diff
@@ -0,0 +1,24 @@
+#include "runtimeid_store.h"
+#include "util.h"
+
+RuntimeIdStore::RuntimeIdStore()
+{
+    m_isIis = IsRunningOnIIS();
+    m_process_runtime_id = GenerateRuntimeId();
+}
+
+void RuntimeIdStore::Generate(AppDomainID app_domain)
+{
+    if (!m_isIis) return;
+
+    std::lock_guard<std::mutex> l(m_mutex);
+    m_appDomainToRuntimeId[app_domain] = GenerateRuntimeId();
+}
+
+const std::string& RuntimeIdStore::Get(AppDomainID app_domain)
+{
+    if (!m_isIis) return m_process_runtime_id;
+
+    std::lock_guard<std::mutex> l(m_mutex);
```

Comment:
I don't really mind, but if you're going to acquire the lock anyway why not generate the id lazily for each appdomain instead of subscribing to the appdomain load callbacks?

Reply:
good point.

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/2474/commits/ecb34d13642094717ad2e65e3fa0d189b80d1b06